storage/src/mmr/mem.rs

```diff
@@ -91,20 +106,20 @@ impl<H: CHasher> Mmr<H> {
     }
 
     /// Return an inclusion proof for the specified element. Returns `ElementPruned` error if the
-    /// requested element is not currently stored by this MMR.
+    /// requested element cannot be proven by this MMR due to pruning.
     pub fn proof(&self, element_pos: u64) -> Result<Proof<H>, Error> {
         self.range_proof(element_pos, element_pos)
     }
 
     /// Return an inclusion proof for the specified range of elements. The range is inclusive of
     /// both endpoints. Returns `ElementPruned` if the requested range is outside the range
-    /// currently stored by this MMR.
+    /// currently provable by this MMR.
     pub fn range_proof(
         &self,
         start_element_pos: u64,
         end_element_pos: u64,
     ) -> Result<Proof<H>, Error> {
-        if start_element_pos != 0 && start_element_pos <= self.oldest_remembered_pos {
+        if self.oldest_remembered_pos != 0 && start_element_pos <= self.oldest_remembered_pos {
```

Comment:
Is there a particular reason that we can't do:
```
if start_element_pos <= self.oldest_remembered_pos {
```

Reply:
Yes, because then you can't do an inclusion proof for element 0.  The other fix would be to allow left-most inclusion proofs as long as the left-most node was a left-sibling.  This gets messy to explain though, so opted to just disallow inclusion proofs for oldest child except for this 0 case.  Open to reconsidering this though.

Reply:
One cleaner design might be to disallow forgetting at node boundaries that are not left-siblings. Right now one can forget at any valid position.

Reply:
So this made me realize that the logic is wrong if we allow pruning to arbitrary points in history, so (for now) I have removed the ability to prune to a specific element and left only forget_max().  I will re-work pruning in a subsequent PR.